hs_geographic_feature_resource/receivers.py

```diff
@@ -312,50 +316,55 @@ def validate_form(request, element_name):
+                res_obj.metadata.coverages.all().delete()
 
-    ori_file_info = res_obj.metadata.originalfileinfo.all().first()
-    if ori_file_info.fileType == "SHP" or ori_file_info.fileType == "ZSHP":
-        del_f_fullname = del_res_fname[del_res_fname.rfind('/')+1:].lower()
-        del_f_name, del_f_ext = os.path.splitext(del_f_fullname)
-        if del_f_ext in [".shp", ".shx", ".dbf"]:
-            # The shp, shx or dbf files cannot be removed.
-            all_file_removed = True
-            one_file_removed = False
-
-            # remove all files in this res right now except for the file user just clicked to remove (hs_core will remove it later)
-            for f in ResourceFile.objects.filter(object_id=res_obj.id):
-                fname = get_resource_file_name(f)
-                if fname != del_res_fname:
-                    delete_resource_file_only(res_obj, f)
-
-        elif del_f_ext == ".prj":
-            originalcoverage_obj = res_obj.metadata.originalcoverage.all().first()
-            res_obj.metadata.update_element('OriginalCoverage', element_id=originalcoverage_obj.id,
-                                            projection_string=UNKNOWN_STR, projection_name=UNKNOWN_STR,
-                                            datum=UNKNOWN_STR, unit=UNKNOWN_STR)
-            res_obj.metadata.coverages.all().delete()
-
-        if one_file_removed:
-            ori_fn_dict = json.loads(ori_file_info.filenameString)
-            if del_f_fullname in ori_fn_dict:
-                del ori_fn_dict[del_f_fullname]
-                res_obj.metadata.update_element('OriginalFileInfo', element_id=ori_file_info.id,
-                                                filenameString=json.dumps(ori_fn_dict))
-        elif all_file_removed:
-            res_obj.metadata.originalfileinfo.all().delete()
-            res_obj.metadata.geometryinformation.all().delete()
-            res_obj.metadata.fieldinformation.all().delete()
-            res_obj.metadata.originalcoverage.all().delete()
-            res_obj.metadata.coverages.all().delete()
+            if one_file_removed:
+                ori_fn_dict = json.loads(ori_file_info.filenameString)
+                if del_f_fullname in ori_fn_dict:
+                    del ori_fn_dict[del_f_fullname]
+                    res_obj.metadata.update_element('OriginalFileInfo', element_id=ori_file_info.id,
+                                                    filenameString=json.dumps(ori_fn_dict))
+            elif all_file_removed:
+                res_obj.metadata.originalfileinfo.all().delete()
+                res_obj.metadata.geometryinformation.all().delete()
+                res_obj.metadata.fieldinformation.all().delete()
+                res_obj.metadata.originalcoverage.all().delete()
+                res_obj.metadata.coverages.all().delete()
+    except Exception as ex:
+        logger.error("pre_delete_file_from_resource: {0}. Error:{1} ".format(res_id, ex.message))
```

Comment:
@zhiyuli I don't think you should add try...except block here just for the purpose of logging the exception message since this will hide all exceptions raised from inside the try block. I think you should remove the try...except block here since if there is an exception (e.g., an iRODS exception being raised from failure to delete the file), it will be raised and populated to the Django website, and there is no need to log the exception into the log.